tests/io/datasets/test_seq.py

```diff
@@ -5,44 +5,49 @@
 import pytest
 
 from libertem.executor.inline import InlineJobExecutor
-from libertem.io.dataset.seq import SEQDataSet
+from libertem.io.dataset.seq import SEQDataSet, _load_xml_from_string, xml_defect_data_extractor
+from libertem.io.dataset.seq import xml_map_sizes, xml_unbinned_map_maker, xml_map_index_selector
+from libertem.io.dataset.seq import xml_defect_coord_extractor, array_cropping
 from libertem.common import Shape
 from libertem.common.buffers import reshaped_view
 from libertem.udf.sumsigudf import SumSigUDF
 from libertem.udf.raw import PickUDF
 from libertem.io.dataset.base import TilingScheme, BufferedBackend, MMapBackend
 
 from utils import get_testdata_path, ValidationUDF
+import defusedxml.ElementTree as ET
 
 try:
     import pims
 except ModuleNotFoundError:
     pims = None
 
-
 SEQ_TESTDATA_PATH = os.path.join(get_testdata_path(), 'default.seq')
 HAVE_SEQ_TESTDATA = os.path.exists(SEQ_TESTDATA_PATH)
 
-pytestmark = pytest.mark.skipif(not HAVE_SEQ_TESTDATA, reason="need .seq testdata")
+needsdata = pytest.mark.skipif(not HAVE_SEQ_TESTDATA, reason="need .seq testdata")
 
 
 @pytest.fixture
+@needsdata
 def default_seq(lt_ctx):
     nav_shape = (8, 8)
+    if HAVE_SEQ_TESTDATA is True:
+        ds = lt_ctx.load(
+            "seq",
+            path=SEQ_TESTDATA_PATH,
+            nav_shape=nav_shape,
+            io_backend=MMapBackend(),
+        )
 
-    ds = lt_ctx.load(
-        "seq",
-        path=SEQ_TESTDATA_PATH,
-        nav_shape=nav_shape,
-        io_backend=MMapBackend(),
-    )
-
-    ds.set_num_cores(4)
-    assert tuple(ds.shape) == (8, 8, 128, 128)
-    return ds
+        ds.set_num_cores(4)
+        assert tuple(ds.shape) == (8, 8, 128, 128)
+        return ds
+    return None
```

Comment:
Test fixtures can't be marked with `skipif` like this - adding `@needsdata` to the individual tests should be enough, the other changes to this fixture should be undone.